benchmarks/overrides_benchmark/README.md

```diff
@@ -0,0 +1,19 @@
+# `__torch_function__` micro-benchmarks
+
+This benchmark suite provides a systemic way to measure the performance of `__torch_function__` overhead.
+
```

Comment:
I'm tempted to add a lot more context here. I can have a go at writing it. 

Interesting:
- Overhead for `torch` functions consuming tensors is on the order of `2 us`. 
- `__torch_function__` should add zero overhead for `Tensor` instances, a small overhead for tensor-like, and on the order of a couple of us for tensor-likes with `__torch_function__` method
- changing the dispatching mechanism may result in small O(100 ns) changes in overhead that are quite difficult to measure well but important.
- etc.